File: src/Test/AckermannFFICheatcode.fs
module Test.AckermannFFICheatcode

let rec ack m n =
    if m = 0 then n + 1
    elif n = 0 then ack (m - 1) 1
    else ack (m - 1) (ack m (n - 1))
let runAckermannFFICheatcode (m: obj) = ack (unbox<int> m) 4 :> obj

File: scratch/compact-guards-integration-20260916/compact/src/Test/AckermannFFICheatcode.cc
#include "purescript.h"
namespace {
int ack(int m, int n) {
    if (m == 0) return n + 1;
    if (n == 0) return ack(m - 1, 1);
    return ack(m - 1, ack(m, n - 1));
}
}
FOREIGN_BEGIN( Test_AckermannFFICheatcode )
exports["runAckermannFFICheatcode"] = [](const boxed& in) -> boxed {
    return ack(unbox<int>(in), 4);
};
FOREIGN_END
